.github/workflows/_binary-test-linux.yml

```diff
@@ -158,28 +163,45 @@ jobs:
           submodules: recursive
           repository: pytorch/builder
           path: builder
+
       - name: Clean pytorch/builder checkout
         run: |
           # Remove any artifacts from the previous checkouts
           git clean -fxd
         working-directory: builder
 
-      - uses: actions/download-artifact@v3
-        name: Download Build Artifacts
+      - name: Check if the job is disabled
+        if: ${{ github.event.workflow_run.head_branch != 'nightly' }}
+        id: filter
+        uses: ./pytorch/.github/actions/filter-test-configs
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          # NB: Use a mock test matrix with a default value here. After filtering, if the
+          # returned matrix is empty, it means that the job is disabled
+          test-matrix: |
+            { include: [
+              { config: "default" },
+            ]}
+
+      - name: Download Build Artifacts
+        if: ${{ github.event.workflow_run.head_branch == 'nightly' || steps.filter.outputs.is-test-matrix-empty == 'False' }}
+        uses: actions/download-artifact@v3
         with:
           name: ${{ inputs.build_name }}
           path: "${{ runner.temp }}/artifacts/"
 
       - name: Install nvidia driver, nvidia-docker runtime, set GPU_FLAG
         uses: pytorch/test-infra/.github/actions/setup-nvidia@main
-        if: ${{ inputs.GPU_ARCH_TYPE == 'cuda' }}
+        if: ${{ inputs.GPU_ARCH_TYPE == 'cuda' && (github.event.workflow_run.head_branch == 'nightly' || steps.filter.outputs.is-test-matrix-empty == 'False') }}
 
       - name: Pull Docker image
+        if: ${{ github.event.workflow_run.head_branch == 'nightly' || steps.filter.outputs.is-test-matrix-empty == 'False' }}
```

Comment:
No need to change it in this PR, but I feel like the ergonomics here would be better if we were checking for a "true" condition instead of a false one, like `steps.filter.outputs.has-test-matrix == 'True'`